src/bin/asr_train.py

```diff
@@ -147,6 +151,19 @@ def main():
         else:
             args.ngpu = 1
 
+    # TODO(nelson) remove in future
+    if (args.train_feat is not None) or \
+       (args.valid_feat is not None) or \
+       (args.train_label is not None) or \
+       (args.valid_label is not None):
+        logging.error(
+            "--train-feat, --valid-feat, --train-label, and valid-label"
+            "options are deprecated, please use --train-json and --valid-json options.")
+        logging.error(
+            "input file format (json) is modified, please redo"
+            "stage 2: Dictionary and Json Data Preparation")
```

Comment:
You may need to change this comment as "stage 1: Feature Generation", as users should redo from stage 1 to dump the features for recognition according to your modifications.

Reply:
You're right, but people have to redo from the stage 1 anyway to finish all experiments.